node/collation-generation/src/lib.rs

```diff
@@ -152,9 +160,7 @@ where
 	type Metrics = Metrics;
 
 	fn start(self, ctx: Context) -> SpawnedSubsystem {
-		let subsystem = CollationGenerationSubsystem { config: None, metrics: self.metrics };
-
-		let future = Box::pin(subsystem.run(ctx));
+		let future = Box::pin(self.run(ctx));
```

Comment:
@coriolinus this is safe since there is no way to construct a subsystem with `Some(config)`.